cartoframes/data/observatory/repository/dataset_repo.py

```diff
@@ -12,50 +14,50 @@ def get_dataset_repo():
 
 class DatasetRepository(EntityRepository):
 
-    id_field = _DATASET_ID_FIELD
+    def __init__(self):
+        super(DatasetRepository, self).__init__(_DATASET_ID_FIELD, _ALLOWED_FILTERS)
 
     def get_by_country(self, iso_code3):
-        return self._get_filtered_entities('country_iso_code3', iso_code3)
+        return self._get_filtered_entities({'country_id': iso_code3})
 
     def get_by_category(self, category_id):
-        return self._get_filtered_entities('category_id', category_id)
+        return self._get_filtered_entities({'category_id': category_id})
 
     def get_by_variable(self, variable_id):
-        return self._get_filtered_entities('variable_id', variable_id)
+        return self._get_filtered_entities({'variable_id': variable_id})
 
     def get_by_geography(self, geography_id):
-        return self._get_filtered_entities('geography_id', geography_id)
+        return self._get_filtered_entities({'geography_id': geography_id})
 
     def get_by_provider(self, provider_id):
-        return self._get_filtered_entities('provider_id', provider_id)
-
-    @classmethod
-    def _map_row(cls, row):
-        return {
-            'id': cls._normalize_field(row, cls.id_field),
-            'name': cls._normalize_field(row, 'name'),
-            'description': cls._normalize_field(row, 'description'),
-            'provider_id': cls._normalize_field(row, 'provider_id'),
-            'category_id': cls._normalize_field(row, 'category_id'),
-            'data_source_id': cls._normalize_field(row, 'data_source_id'),
-            'country_iso_code3': cls._normalize_field(row, 'country_iso_code3'),
-            'language_iso_code3': cls._normalize_field(row, 'language_iso_code3'),
-            'geography_id': cls._normalize_field(row, 'geography_id'),
-            'temporal_aggregation': cls._normalize_field(row, 'temporal_aggregation'),
-            'time_coverage': cls._normalize_field(row, 'time_coverage'),
-            'update_frequency': cls._normalize_field(row, 'update_frequency'),
-            'version': cls._normalize_field(row, 'version'),
-            'is_public_data': cls._normalize_field(row, 'is_public_data'),
-            'summary_jsonb': cls._normalize_field(row, 'summary_jsonb')
-        }
+        return self._get_filtered_entities({'provider_id': provider_id})
 
     @classmethod
     def _get_entity_class(cls):
         from cartoframes.data.observatory.dataset import Dataset
         return Dataset
 
-    def _get_rows(self, field=None, value=None):
-        return self.client.get_datasets(field, value)
+    def _get_rows(self, filters=None):
+        return self.client.get_datasets(filters)
```

Comment:
If the filter contains the `GEOGRAPHY_FILTER` I guess we should do a join with the geographies table (since we want to filter by the geography name).

So to support this navigation:

```python
catalog.country('usa').category('demographics').geography('USA Census Block Group').datasets
```

We would need something like this:

```sql
SELECT distinct d.* FROM datasets_public d LEFT JOIN geographies_public g ON g.id = d.geography_id where category_id = 'demographics' and d.country_id = 'usa' and g.name = 'USA Census Block Group'
```

For the rest of the cases it's fine to filter by the IDs (country, category, etc.) in the `datasets` view.

Reply:
I talked with @alasarr and we the idea is to filter by geography id as with the rest of entities.